av.c

```diff
@@ -449,6 +449,113 @@ Perl_av_make(pTHX_ SSize_t size, SV **strp)
     return av;
 }
 
+/*
+=for apidoc newAVav
+
+Creates a new AV and populates it with values copied from an existing AV.  The
+new AV will have a reference count of 1, and will contain newly created SVs
+copied from the original SV.  The original source will remain unchanged.
+
+Perl equivalent: C<my @new_array = @existing_array;>
+
+=cut
+*/
+
+AV *
+Perl_newAVav(pTHX_ AV *oav)
+{
+    PERL_ARGS_ASSERT_NEWAVAV;
+
+    if(UNLIKELY(!oav))
+        return newAV();
+
+    if(LIKELY(!SvRMAGICAL(oav))) {
+        return av_make(av_count(oav), AvARRAY(oav));
+    }
+
+    AV *ret = newAV();
+
+    /* avoid ret being leaked if croak when calling magic below */
```

Comment:
This feels like a rather awkward way to do this. Can't we just mortalize it, and then ref-increment it on success?

Reply:
Hmm that might work too. I just stole this structure from the `Perl_av_make` function right above it. It seemed simple enough.

Reply:
Fair enough